pkg/city/city.go

```diff
@@ -29,3 +37,40 @@ func (c *City) GetStopsByID() map[uint64]*GraphNode {
 func (c *City) GetTimeBounds() TimeBounds {
 	return c.cityData.GetTimeBounds()
 }
+
+func (c *City) GetLinesForStop(stopID uint64, chipPerRowSize int) []string {
+	if lines, ok := c.linesByStopID[stopID]; ok {
+		for start := 0; start < len(lines); start += chipPerRowSize {
+			end := start + chipPerRowSize
+			if end > len(lines) {
+				end = len(lines)
+			}
+			slices.Reverse(lines[start:end])
+		}
+		return lines
+	} else {
+		return []string{}
+	}
+}
+
+func (c *City) GetArrivalsForStop(stopID uint64, currentTime uint, numberOfArrivals int) (upcoming []Arrival) {
+	upcoming = []Arrival{}
+	for _, arrival := range c.arrivalsByStopID[stopID] {
+		if arrival.ETA+30 >= currentTime {
+			diff := arrival.ETA - currentTime
+			var eta uint = 0
```

Comment:
Please add some line spacing between sections of your code for better readability. Also the variable will be initialized to zero by default so no need to be explicit.
```suggestion

			var eta uint
```